pkg/kubernetes/scripts/virtual-machines/components/createVmButton.jsx

```diff
@@ -0,0 +1,117 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+
+import { connect } from 'react-redux'
+import React, { PropTypes } from "react";
+import DialogPattern from 'cockpit-components-dialog.jsx';
+import CreateVmDialog from './createVmDialog.jsx'
+
+import { mouseClick } from '../utils.jsx'
+
+const _ = cockpit.gettext;
+
+
+class CreateVmButton extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.createVmDialog = this.createVmDialog.bind(this);
+        this.createVm = this.createVm.bind(this);
+    }
+
+
+    createVm(resource) {
+        const deferred = cockpit.defer();
+        const promise = this.props.kubeMethods.create(resource, this.props.kubeLoader.limits.namespace);
```

Comment:
I think we should externalize such processing to sagas.
I'm working on implementation of lifecycle actions, we can agree on code design there, if it works.

Reply:
I agree that this isn't UI related code and can be externalized. I have serous doubts that sagas are the best place for that since:
* Sagas are intended for asynchronous flows that change redux store - store isn't touched there.
* Sagas use clumsy generator notation instead of cleaner promise notation.
* Sagas are invoked using actions and action-creatores which I find to be pure boilerplate provided store is not touched.
* Sagas plague code with effects which makes it less readable (yet easier to unit-test).

So I'd suggest to create new separate file/module holding backend-ish flows in from of (potentially async) functions and providing them dependencies like `kubeMethods` in closure.

Reply:
The "middleware" is implemented within https://github.com/cockpit-project/cockpit/pull/8470.
Depending on what is merged first, we can potentially refactor later.
I would not block this PR on that.